### ISSUE TITLE
`blank-lines-top-level` (`E302`) - conflicts with formatter on `.pyi` files

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#10039]
```py
# foo.pyi
class Foo: ...
class Bar: ... # error: Expected 2 blank lines, found 0
```

when running `ruff format` on this file, it removes the additional blank lines that `E302` expects

**Comment by MichaReiser** [comment]
Nice find. We should review all the E3* rules and verify that they enforce at most one blank lines for pyi files rather than 2

**Comment by KotlinIsland** [comment]
I've also detected E305 blank-lines-after-function-or-class also conflicts
